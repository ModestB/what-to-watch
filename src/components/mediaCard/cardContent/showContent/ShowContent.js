import React from "react";

import useRoute from "../../../../hooks/useRoute";

import classes from "./ShowContent.module.scss";

export const ShowContent = (props) => {
  const { isSinglePage } = useRoute();

  let content = null;
  let cardTitle = null;
  let cardText = null;
  let date = null;

  if (!isSinglePage) {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle.length > 40
          ? props.showTitle.substring(0, 40) + "..."
          : props.showTitle}
      </div>
    );
  } else {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle}
      </div>
    );
  }

  if (props.showOverview && !isSinglePage) {
    cardText = (
      <p className={`${classes.description}`}>
        {props.showOverview.substring(0, 100) + "..."}
      </p>
    );
  } else {
    cardText = (
      <p
        className={`${[classes.description, classes.descriptionOverview].join(
          " "
        )} customScroll`}
      >
        {props.showOverview}
      </p>
    );
  }

  if (props.mediaType === "movie") {
    date = <p className={`${classes.date}`}>Released Date: {props.showDate}</p>;
  } else {
    date = <p className={`${classes.date}`}>Air Date: {props.showDate}</p>;
  }

  content = (
    <React.Fragment>
      {cardTitle}
      {cardText}
      {date}
    </React.Fragment>
  );
  return content;
};

export default ShowContent;
