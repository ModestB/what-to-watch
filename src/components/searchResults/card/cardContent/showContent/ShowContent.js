import React from "react";

import { connect } from "react-redux";

import classes from "./ShowContent.module.scss";

export const ShowContent = (props) => {
  let content = null;
  let cardTitle = null;
  let cardText = null;
  let date = null;

  if (!props.displaySinglePage) {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle.length > 40
          ? props.showTitle.substring(0, 40) + "..."
          : props.showTitle}
      </div>
    );
  } else {
    cardTitle = (
      <div className={`${classes.title}`} title={props.showTitle}>
        {props.showTitle}
      </div>
    );
  }

  if (props.showOverview && !props.displaySinglePage) {
    cardText = (
      <p className={`${classes.description}`}>
        {props.showOverview.substring(0, 130) + "..."}
      </p>
    );
  } else {
    cardText = (
      <p
        className={`${[classes.description, classes.descriptionOverview].join(
          " "
        )} customScroll`}
      >
        {props.showOverview}
      </p>
    );
  }

  if (props.mediaType === "movie") {
    date = <p className={`${classes.date}`}>Released Date: {props.showDate}</p>;
  } else {
    date = <p className={`${classes.date}`}>Air Date: {props.showDate}</p>;
  }

  content = (
    <React.Fragment>
      {cardTitle}
      {cardText}
      {date}
    </React.Fragment>
  );
  return content;
};

const mapStateProps = (state) => {
  return {
    displaySinglePage: state.displaySinglePage,
  };
};

export default connect(
  mapStateProps,
  null
)(ShowContent);
