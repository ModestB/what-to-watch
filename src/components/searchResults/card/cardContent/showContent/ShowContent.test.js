import React from "react";

import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { ShowContent } from "./ShowContent";

configure({ adapter: new Adapter() });

describe("<ShowContent />", () => {
  let wrapper;
  let props = {
    showTitle: "Star Wars: The Rise of Skywalker Skywalker",
    showOverview:
      "The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.",
    showDate: "2019-12-18",
  };

  beforeEach(() => {
    wrapper = shallow(<ShowContent {...props} />);
  });

  it("should match snapshot", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match snapshot mediaTyoe = movie", () => {
    wrapper.setProps({ mediaType: "movie" });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should match snapshot displaySinglePage = true", () => {
    wrapper.setProps({ displaySinglePage: true });
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("should ellipse title", () => {
    expect(wrapper.find(".title").text()).toHaveLength(43);
    expect(
      wrapper
        .find(".title")
        .text()
        .endsWith("...")
    ).toEqual(true);
    expect(wrapper.find(".title").text()).not.toEqual(props.showTitle);
  });

  it("should not ellipse title", () => {
    wrapper.setProps({ showTitle: "Star Wars: The Rise of Skywalker" });
    expect(wrapper.find(".title").text()).toEqual(
      "Star Wars: The Rise of Skywalker"
    );
  });

  it("should ellipse overview", () => {
    expect(wrapper.find(".description").text()).toHaveLength(133);
    expect(
      wrapper
        .find(".description")
        .text()
        .endsWith("...")
    ).toEqual(true);
    expect(wrapper.find(".description").text()).not.toEqual(props.showOverview);
  });

  it("should not ellipse overview", () => {
    wrapper.setProps({ displaySinglePage: true });
    expect(wrapper.find(".description").text()).toEqual(props.showOverview);
  });

  it("should start with Air Date", () => {
    expect(
      wrapper
        .find(".date")
        .text()
        .startsWith("Air Date:")
    ).toEqual(true);
  });

  it("should start with Released Date", () => {
    wrapper.setProps({ mediaType: "movie" });
    expect(
      wrapper
        .find(".date")
        .text()
        .startsWith("Released Date:")
    ).toEqual(true);
  });
});
