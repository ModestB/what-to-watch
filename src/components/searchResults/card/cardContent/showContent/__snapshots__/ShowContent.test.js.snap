// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowContent /> should match snapshot 1`] = `
"<_default>
  <div className=\\"title\\" title=\\"Star Wars: The Rise of Skywalker Skywalker\\">
    Star Wars: The Rise of Skywalker Skywalk...
  </div>
  <p className=\\"description\\">
    The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power an...
  </p>
  <p className=\\"date\\">
    Air Date: 
    2019-12-18
  </p>
</_default>"
`;

exports[`<ShowContent /> should match snapshot displaySinglePage = true 1`] = `
"<_default>
  <div className=\\"title\\" title=\\"Star Wars: The Rise of Skywalker Skywalker\\">
    Star Wars: The Rise of Skywalker Skywalker
  </div>
  <p className=\\"description descriptionOverview customScroll\\">
    The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.
  </p>
  <p className=\\"date\\">
    Air Date: 
    2019-12-18
  </p>
</_default>"
`;

exports[`<ShowContent /> should match snapshot mediaTyoe = movie 1`] = `
"<_default>
  <div className=\\"title\\" title=\\"Star Wars: The Rise of Skywalker Skywalker\\">
    Star Wars: The Rise of Skywalker Skywalk...
  </div>
  <p className=\\"description\\">
    The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power an...
  </p>
  <p className=\\"date\\">
    Released Date: 
    2019-12-18
  </p>
</_default>"
`;
